pkg/vsphere/vm/vm.go

```diff
@@ -281,7 +284,37 @@ func (vm *VirtualMachine) DeleteExceptDisks(ctx context.Context) (*object.Task,
 		return nil, err
 	}
 
-	return vm.Destroy(op)
+	info, err := vm.WaitForResult(op, func(ctx context.Context) (tasks.Task, error) {
+		return vm.Destroy(ctx)
+	})
+
+	if err == nil {
+		return info, nil
+	}
+
+	if isMethodDisabledError(ctx, err) {
+		vm.EnableDestroy(ctx)
+
+		return vm.WaitForResult(op, func(ctx context.Context) (tasks.Task, error) {
+			return vm.Destroy(ctx)
+		})
+	}
+
+	return nil, err
+}
+
+func isMethodDisabledError(ctx context.Context, err error) bool {
+	op := trace.FromContext(ctx, "isMethodDisabledError")
+
+	if soap.IsSoapFault(err) {
+		vimFault := soap.ToSoapFault(err).VimFault()
+		op.Debugf("Error type: %s", reflect.TypeOf(vimFault))
```

Comment:
I don't think this method needs the logging and it's not consistent with the `isConcurrentAccess` check. We report the error in the caller of `DeleteExceptDisks` if it's not a method disabled and we should be logging in `EnableDestroy`.
Not a strong preference on my part.

Reply:
I will move all the error type checks to tasks package and make them all consistent. 
If it's not method disabled error, we report it back to DeleteExceptDisks, and the latter report back to the caller for retry. And then the caller logs the error message if retry fails.